### PR TITLE
Update to 0.10.0

### PR DESCRIPTION
**Pull request by sraturi**
updated gradle to 4.10.2
updated rightmesh plugin to 1.8.1
updated rightmesh lib to 0.10.0
updated the meshId constructor
ran checkstyle

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -66,11 +47,15 @@ android {
         targetCompatibility 1.8
         sourceCompatibility 1.8
     }
+    lintOptions {
+        abortOnError false
```
Comment (FrazerSeymour) [inline app/build.gradle:46]: I don't _think_ this should be necessary. Were you having lint issues?

File: app/src/main/java/io/left/ripple/MainActivity.java [hunk 2]
```diff
@@ -227,7 +227,13 @@ private void receiveColourMessage(RightMeshEvent rme) {
         int separatorIndex = dataString.indexOf(':');
 
         // Transmit the message forward if this device is not the intended final recipient.
-        MeshId recipient = new MeshId(dataString.substring(0, separatorIndex).getBytes());
+        MeshId recipient = null;
+        try {
+            recipient = MeshId.fromString(dataString.substring(0,separatorIndex));
+        } catch (RightMeshException e) {
```
Comment (FrazerSeymour) [inline app/src/main/java/io/left/ripple/MainActivity.java:236]: You might want to return here instead of continuing. Looks like the next line will NPE if it tries to call `recipient.equals()`.

**Review by FrazerSeymour — CHANGES_REQUESTED** [review]

File: app/build.gradle (hunk 1 above)
Comment (sraturi) [reply app/build.gradle:46]: yeah. although now that im thinking i can run checkstyle over it and see if that  solves it.

Comment (sraturi) [reply app/build.gradle:46]: still fails. either this or I guess I can try fixing the lints.

Comment (FrazerSeymour) [reply app/build.gradle:46]: Looks like it's related to the leaky EthereumJ dependency. We'll have to stick with the bandaid fix for now.

**Review by FrazerSeymour — APPROVED** [review]

**Review by raycho — APPROVED** [review]

File: app/src/main/java/io/left/ripple/MainActivity.java [hunk 3]
```diff
@@ -36,6 +36,7 @@
     // Interface object for the RightMesh library.
     AndroidMeshManager androidMeshManager;
 
+    //update this to your  assigned mesh port
```
Comment (conbrad) [inline app/src/main/java/io/left/ripple/MainActivity.java]: Minor: too many spaces

File: app/build.gradle [hunk 4]
```diff
@@ -1,63 +1,44 @@
...
-            mavenCentral()
-        }
-        dependencies {
-            classpath 'io.left.rightmesh:rightmesh-plugin:1.6'
-        }
-    }
 }
-apply plugin: 'io.left.rightmesh.rightmesh-plugin'
+
 preBuild.dependsOn("rightmesh")
```
Comment (conbrad) [inline app/build.gradle]: Minor: Why so much white space?

File: app/build.gradle [hunk 5]
```diff
@@ -66,13 +47,17 @@ android {
         targetCompatibility 1.8
         sourceCompatibility 1.8
     }
+    lintOptions {
+        abortOnError false
```
Comment (conbrad) [inline app/build.gradle:46]: ~~Why is this needed now?~~ Nevermind, saw the discussion above.

**Review by conbrad — CHANGES_REQUESTED** [review]

**Review by raycho — APPROVED** [review]

File: app/src/main/java/io/left/ripple/MainActivity.java (hunk 3 above)
Comment (FrazerSeymour) [reply app/src/main/java/io/left/ripple/MainActivity.java]: You can actually use GitHub suggestions for this, which is pretty cool:
```suggestion
    / Update this to your assigned mesh port.
```

Comment (FrazerSeymour) [reply app/src/main/java/io/left/ripple/MainActivity.java]: So now he can click "accept" and it just auto-commits it.

Comment (FrazerSeymour) [reply app/src/main/java/io/left/ripple/MainActivity.java]: I wonder if you can suggest on a suggestion. I guess you'd just make a different suggestion in the same thread.
```suggestion
    // Ah, yeah, just puts you into the same line.
```

File: app/build.gradle (hunk 4 above)
Comment (sraturi) [reply app/build.gradle]: removed.

File: app/src/main/java/io/left/ripple/MainActivity.java (hunk 3 above)
Comment (conbrad) [reply app/src/main/java/io/left/ripple/MainActivity.java]: 😮

**Review by conbrad — APPROVED** [review]
